crates/ironrdp-acceptor/src/connection.rs

```diff
@@ -123,6 +123,11 @@ impl Acceptor {
         }
     }
 
+    /// # Panics
+    ///
+    /// If state is not [AcceptorState::SecurityUpgrade].
+    ///
+    /// If transition from [AcceptorState::SecurityUpgrade] to next state failed.
```

Comment:
suggestion: Do not document the `debug_assert!`, because it’s just for debugging and panic is not intended

Reply:
I should silence the lint instead, right?

Reply:
Done